gcloud/bigtable/column_family.py

```diff
@@ -177,8 +177,23 @@ class ColumnFamily(object):
 
     :type table: :class:`Table <gcloud.bigtable.table.Table>`
     :param table: The table that owns the column family.
+
+    :type gc_rule: :class:`GarbageCollectionRule`
+    :param gc_rule: (Optional) The garbage collection settings for this
+                    column family.
     """
 
-    def __init__(self, column_family_id, table):
+    def __init__(self, column_family_id, table, gc_rule=None):
         self.column_family_id = column_family_id
         self._table = table
+        self.gc_rule = gc_rule
+
+    def __eq__(self, other):
+        if not isinstance(other, self.__class__):
+            return False
+        return (other.column_family_id == self.column_family_id and
+                other._table == self._table and
+                other.gc_rule == self.gc_rule)
+
+    def __ne__(self, other):
+        return not self.__eq__(other)
```

Comment:
I'm talking about the simplified case, where `__eq__` always returns a sane `bool`:

```python
>>> class Foo(object):
...     def __init__(self, value):
...         self.value = value
...     def __eq__(self, other):
...         return (self.__class__ is other.__class__ and
...                 self.value == other.value)
... 
>>> bar = Foo('bar')
>>> baz = Foo('baz')
>>> 
>>> bar == baz
False
>>> bar != baz
True
>>> bar == bar
True
>>> bar != bar
False
```

I've got no theory for what your example is triggering, but since it is absurd to start with...